### ISSUE TITLE
looks like compatibility.process_entries() is not scrubbing previous energy_adjustments

### ISSUE THREAD

**Issue by computron** [materialsproject/pymatgen#1922]
**Describe the bug**
pymatgen.entries.compatibility.Compatibility.process_entries() is supposed to be clearing out the previous energy adjustments on an entry (the  "clean" parameter is by default True). Doesn't look like it's doing it.

**To Reproduce**
- Load a ComputedEntry with existing energy corrections
- Run MaterialsProjectCompatibility.process_entries() on that entry - which should in principle change nothing
- watch the energy adjustments change

**Can fix by**
Making the following patch ~line 474 of compatibility.py
```
            if clean:
                entry.energy_adjustments = []
```
rather than the weird convoluted code that tries to individually remove each adjustment (why?)

**Comment by computron** [comment]
Actually closing for now, this looks to be something to do with the way an entry had been serialized in a database and then deserialized. I can't reproduce using more straightforward code. Although I would suggest that the code above is probably more robust than what's there as it works for the example below as well as what I was working with. Feel free to close whenever desired.

Here is the code that confirms that things are actually working as intended:

```
from pymatgen import MPRester
from pymatgen.entries.compatibility import MaterialsProjectCompatibility

mpr = MPRester()
mpc = MaterialsProjectCompatibility()

entry = mpr.get_entries("mp-19042")[0]
print(len(entry.energy_adjustments), entry.correction)

entry2 = mpc.process_entry(entry)  # should clean the previous one
print(len(entry2.energy_adjustments), entry2.correction)

# It seems to do so successfully.
```

**Comment by mkhorton** [comment]
Not against simplifying the `clean` method regardless, I'm not sure what the current logic is. @rkingsbury, thoughts?

**Comment by rkingsbury** [comment]
Just to make sure - @computron did you pull in the latest pymatgen in which I changed the default behavior of `clean`? We had it set to `False` which did indeed cause a lot of unexpected behavior. See https://github.com/materialsproject/pymatgen/pull/1919

No objection to simplifying the code a bit; I think the for loop might be a leftover from a previous iteration on this idea in which we were trying to make certain adjustments 'immune' to cleaning.

I know that there can be some non-obvious behavior with entries that were serialized before we implemented energy adjustments. I tried to handle all of those cases (and have tests in pymatgen that should catch them) but if you find a reproducible case please reopen.

**Comment by computron** [comment]
To follow up - I did have the most recent version of the code with the default being True.

I think what happened in my case is as follows:
- A ComputedEntry was exported to MongoDB
- ComputedEntry.from_dict() was used to load it back from MongoDB
- For some reason, the deserialization of energy adjustments probably did not occur in a great manner. Perhaps it got deserialized to some object that didn't have a great equals() method or something.

So then the ``clean()`` implementation, which calls list.remove(), was clearly running (I tested this) but failing to actually remove anything. Again, probably because list.remove() requires a proper equals() method to work properly.

Style-wise, I also find it problematic to be removing entries from a list while looping over those same entries:
(current code)
```
            if clean:
                for ea in entry.energy_adjustments:
                    entry.energy_adjustments.remove(ea)
```
(proposed)
```
            if clean:
                entry.energy_adjustments = []
```
